### PR TITLE
Add cluster autoscaler and helm deployment

### PR DESCRIPTION
**Pull request by janavenkat**
Already added tags in place in autoscaling group

k8s.io/cluster-autoscaler/&lt;cluster-name&gt; | owned
-- | --
k8s.io/cluster-autoscaler/enabled | TRUE

### CODE REVIEW

File: eks/cluster-autoscaler.tf [hunk 1]
```diff
@@ -0,0 +1,29 @@
+resource "helm_release" "cluster_autoscaler" {
+  name = "cluster_autoscaler"
+  repository = "https://kubernetes.github.io/autoscaler"
+  chart      = "cluster-autoscaler"
+  // App version should match the kubernetes version with in the chart
+  // chart version 9.9.2 support kubernetes version 1.20
+  // to get the chart version `helm search repo autoscaler/cluster-autoscaler --versions`
+  version = "9.9.2"
```
Comment (abhiyerra) [inline eks/cluster-autoscaler.tf:8]: Can you make this a variable in variables.tf?

```
cluster_autoscaler_version
```

File: eks/cluster-autoscaler.tf [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  values = [<<EOF
+autoDiscovery:
+  clusterName: ${var.environment_name}
+  awsRegion: us-east-1
+  tags:
+    - k8s.io/cluster-autoscaler/enabled
+    - k8s.io/cluster-autoscaler/{{ .Values.autoDiscovery.clusterName }}
```
Comment (abhiyerra) [inline eks/cluster-autoscaler.tf:15]:
```
${var.environement_name}
```

File: eks/iam.tf [hunk 3]
```diff
@@ -262,3 +262,44 @@ resource "aws_iam_policy" "efs_policy" {
...
+
+module "iam_assumable_role_cluster_autoscaler" {
+  source           = "terraform-aws-modules/iam/aws//modules/iam-assumable-role-with-oidc"
+  version          = "3.6.0"
+  create_role      = true
+  role_name        = "efs-driver"
+  provider_url     = replace(aws_iam_openid_connect_provider.cluster.url, "https://", "")
+  role_policy_arns = [aws_iam_policy.cluster_autoscaler_policy.arn]
+  # namespace and service account name
+  oidc_fully_qualified_subjects = ["system:serviceaccount:kube-system:cluster-autoscaler"]
```
Comment (abhiyerra) [inline eks/iam.tf:275]: Is there a way to reference this?

File: eks/iam.tf [hunk 4]
```diff
@@ -262,3 +262,44 @@ resource "aws_iam_policy" "efs_policy" {
 }
 EOF
 }
+
+
+module "iam_assumable_role_cluster_autoscaler" {
+  source           = "terraform-aws-modules/iam/aws//modules/iam-assumable-role-with-oidc"
+  version          = "3.6.0"
+  create_role      = true
+  role_name        = "efs-driver"
```
Comment (abhiyerra) [inline eks/iam.tf:271]: This was not changed

**Review by abhiyerra — CHANGES_REQUESTED** [review]

File: eks/iam.tf (hunk 3 above)
Comment (janavenkat) [reply eks/iam.tf:275]: we can make a variable like `ca-serice-account-name` and use it here and helm chart

File: eks/iam.tf (hunk 4 above)
Comment (janavenkat) [reply eks/iam.tf:271]: good one didn't noticed

File: eks/iam.tf (hunk 3 above)
Comment (janavenkat) [reply eks/iam.tf]: added the variable for service account

File: eks/cluster-autoscaler.tf [hunk 5]
```diff
@@ -0,0 +1,26 @@
+resource "helm_release" "cluster_autoscaler" {
```
Comment (abhiyerra) [inline eks/cluster-autoscaler.tf:1]: Can you rename this file cluster_autoscaler.tf

File: eks/iam.tf [hunk 6]
```diff
@@ -262,3 +262,44 @@ resource "aws_iam_policy" "efs_policy" {
...
+
+module "iam_assumable_role_cluster_autoscaler" {
+  source           = "terraform-aws-modules/iam/aws//modules/iam-assumable-role-with-oidc"
+  version          = "3.6.0"
+  create_role      = true
+  role_name        = "cluster-autoscaler"
+  provider_url     = replace(aws_iam_openid_connect_provider.cluster.url, "https://", "")
+  role_policy_arns = [aws_iam_policy.cluster_autoscaler_policy.arn]
+  # namespace and service account name
+  oidc_fully_qualified_subjects = ["system:serviceaccount:kube-system:${var.cluster_autoscaler_service_account_name}"]
```
Comment (abhiyerra) [inline eks/iam.tf:275]: Please put this on a new line:

```
oidc_fully_qualified_subjects = [
    "system:serviceaccount:kube-system:${var.cluster_autoscaler_service_account_name}"
]
```

File: eks/variables.tf [hunk 7]
```diff
@@ -6,6 +6,17 @@ variable "cluster_version" {
...
 }
 
+// App version should match the kubernetes version with in the chart
+// chart version 9.9.2 support kubernetes version 1.20
+// to get the chart version `helm search repo autoscaler/cluster-autoscaler --versions`
+variable "cluster_autoscaler_version" {
+  default = "9.9.2"
+}
+
+variable "cluster_autoscaler_service_account_name" {
```
Comment (abhiyerra) [inline eks/variables.tf:16]: Change this to `cluster_autoscaler_name`

File: eks/cluster-autoscaler.tf [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+  name = "cluster_autoscaler"
```
Comment (abhiyerra) [inline eks/cluster-autoscaler.tf:2]: Change this to `var.cluster_autoscaler_name`

File: eks/cluster-autoscaler.tf [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+  repository = "https://kubernetes.github.io/autoscaler"
+  chart      = "cluster-autoscaler"
+  version = var.cluster_autoscaler_version
+  values = [<<EOF
+autoDiscovery:
+  clusterName: ${var.environment_name}
+  awsRegion: us-east-1
```
Comment (abhiyerra) [inline eks/cluster-autoscaler.tf:9]: This needs to be a variable. Figure out the region of the cluster and add it.

**Review by abhiyerra — CHANGES_REQUESTED** [review]

**Pull request merged**